agent/agent.go

```diff
@@ -625,6 +625,15 @@ func (a *Agent) Start(ctx context.Context) error {
 	if c.ServerMode {
 		serverLogger := a.baseDeps.Logger.NamedIntercept(logging.ConsulServer)
 
+		// TODO: maybe this is called too early?
+		if err := a.checkServerLastSeen(); err != nil {
+			// TODO: log a  bunch of times first?
```

Comment:
I believe logging loudly was called out in the requirements, can we resolve this TODO by adding that?

Reply:
The requirements state:

>In this state, they log loudly, wait around for 60 seconds and then quit.

I'm not sure what waiting for 60s before exiting achieves as opposed to erroring and exiting immediately. Maybe @banks can chime in on this? I can definitely just add a 60s ticker/loop that logs, if that's the functionality we want to go with.

Reply:
The rationale was two fold:

 1. Just in case someone is not running Consul with a well-behaved supervisor, if we commit immediately it could just end up in a crash loop.
 2. Even if the supervisor is good and throttles it, there is not benefit to crashing quickly since this is a terminal condition. If we just log once and exit it seems much more likely that the operator might have to spend a while looking closely at logs to work out why - there will be one error line saying why surrpoundied by maybe hundreds of lines of shutdoan/startup noise.

So given that it's terminal anyway it seems like a nicer behaviour to force it to be a slow crash loop ourselves - stay up for some time and log more than once so that an operator will immediately see the error line in logs not a see of startup/crash noise.

I'll leave it to you to decide if thats important but it seemed like a better operator UX and should only require a few lines of code or so to loop, log and sleep before finally exiting?

Reply:
https://github.com/hashicorp/consul/pull/17171/commits/248055683779771e766382df34d3263d7fc94808